From c0d0e9fd94e95bf44474cf04b1125a211bd2bfb3 Mon Sep 17 00:00:00 2001
From: David Pacheco <dap@joyent.com>
Date: Thu, 9 Feb 2017 10:25:01 -0800
Subject: [PATCH] joyent/manta-mlive#6 mlive could be less frenetic about slow
 requests joyent/manta-mlive#7 mlive verbose logs should go to stderr Reviewed
 by: Joshua M. Clulow <jmc@joyent.com> Approved by: Joshua M. Clulow
 <jmc@joyent.com>

---
 bin/mlive | 19 +++++--------------
 1 file changed, 5 insertions(+), 14 deletions(-)

diff --git a/bin/mlive b/bin/mlive
index 959d6dd..043244f 100755
--- a/bin/mlive
+++ b/bin/mlive
@@ -126,7 +126,8 @@ function main()
 
 	args.ml_log = new mod_bunyan({
 	    'name': mlArg0,
-	    'level': process.env['LOG_LEVEL'] || 'warn'
+	    'level': process.env['LOG_LEVEL'] || 'warn',
+	    'stream': process.stderr
 	});
 
 	args.ml_manta = mod_manta.createBinClient({ 'log': args.ml_log });
@@ -398,34 +399,24 @@ LivenessTester.prototype.ensureObject = function (path, callback)
 };
 
 /*
- * Invoked frequently (10Hz by default) to fire off new requests.
+ * Invoked frequently (20Hz by default) to fire off new requests.
  */
 LivenessTester.prototype.tick = function ()
 {
 	var args = this.mlt_args;
 	var self = this;
-	var now, rq, method, rqargs;
+	var rq, method, rqargs;
 
 	mod_assertplus.ok(this.mlt_noutstanding <= args.ml_maxrqs);
 	if (this.mlt_noutstanding == args.ml_maxrqs) {
 		if (this.mlt_corked === null) {
 			this.mlt_corked = new Date();
-			fprintf(args.ml_error, '%s: corked (at %d requests)\n',
-			    this.mlt_corked.toISOString(),
-			    this.mlt_noutstanding);
 		}
 
 		return;
 	}
 
-	if (this.mlt_corked !== null) {
-		now = new Date();
-		fprintf(args.ml_error, '%s: uncorked (was corked for %d ms)\n',
-		    now.toISOString(),
-		    now.getTime() - this.mlt_corked.getTime());
-		this.mlt_corked = null;
-	}
-
+	this.mlt_corked = null;
 	this.mlt_noutstanding++;
 	rq = this.mlt_requests[this.mlt_which++];
 	this.mlt_which %= this.mlt_requests.length;
-- 
2.21.0

